src/minerva/services/service_manager.py

```
@@ -478,6 +482,228 @@ def search_by_alias(self, alias: str, directory: str | None = None) -> list[str]
+            except Exception as e:
+                logger.warning("Could not clear existing database: %s", e)
+
+        # Get the actual embedding dimension (shape[1] for 2D array)
+        embedding_dim = (
+            sample_embedding.shape[1]
+            if sample_embedding.ndim == 2
+            else len(sample_embedding)
+        )
+        indexer.initialize_schema(embedding_dim)
+
+    def build_vector_index(
```

Pylint (3.3.7)

[refactor] Too many local variables (24/15)

(R0914)